src/DssSpell.t.sol

```diff
@@ -1856,149 +2115,387 @@ contract DssSpellTest is DSTest, DSMath {
+        vat.frob("ETH-A", address(this), address(this), address(this), int256(ilkAmt), int256(mul(ilkAmt, spot) / rate));
+        (, rate, spot,,) = vat.ilks("BAL-A");
+        vat.frob("BAL-A", address(this), address(this), address(this), int256(ilkAmt), int256(mul(ilkAmt, spot) / rate));
+        (, rate, spot,,) = vat.ilks("LINK-A");
+        vat.frob("LINK-A", address(this), address(this), address(this), int256(ilkAmt), int256(mul(ilkAmt, spot) / rate));
+
+        hevm.warp(block.timestamp + 1);
+        jug.drip("ETH-A");
+        jug.drip("BAL-A");
+        jug.drip("LINK-A");
+
+        uint256 auctionIdETHA = flipETHA.kicks() + 1;
+        uint256 auctionIdBALA = flipBALA.kicks() + 1;
+        uint256 auctionIdLINKA = clipLINKA.kicks() + 1;
+
+        cat.bite("ETH-A", address(this));
+        cat.bite("BAL-A", address(this));
+        dog.bark("LINK-A", address(this), address(this));
+
+        assertEq(flipETHA.kicks(), auctionIdETHA);
+        assertEq(flipBALA.kicks(), auctionIdBALA);
+        assertEq(clipLINKA.kicks(), auctionIdLINKA);
+
+        hevm.store(
+            address(end),
+            keccak256(abi.encode(address(this), uint256(0))),
+            bytes32(uint256(1))
+        );
+        assertEq(end.wards(address(this)), 1);
 
-        tellSpell.schedule();
+        end.cage();
+        end.cage("ETH-A");
+        end.cage("BAL-A");
+        end.cage("LINK-A");
 
-        uint256 castTime = block.timestamp + pause.delay();
-        hevm.warp(castTime);
-        (, , uint48 tau, uint48 tocPre) = oracle.ilks(ilk);
-        assertEq(uint256(tocPre), 0);
-        assertTrue(oracle.good(ilk));
-        tellSpell.cast();
-        (, , , uint48 tocPost) = oracle.ilks(ilk);
-        assertEq(uint256(tocPost), block.timestamp);
-        if (tau == 0) {
-            assertTrue(!oracle.good(ilk));
-        } else {
-            assertTrue(oracle.good(ilk));
-        }
-        hevm.warp(block.timestamp + 600);
-        assertTrue(!oracle.good(ilk));
+        (,, address usr,,,,,) = flipETHA.bids(auctionIdETHA);
+        assertTrue(usr != address(0));
+        (,, usr,,,,,) = flipBALA.bids(auctionIdBALA);
+        assertTrue(usr != address(0));
+        (,,, usr,,) = clipLINKA.sales(auctionIdLINKA);
+        assertTrue(usr != address(0));
+
+        end.skip("ETH-A", auctionIdETHA);
+        end.skip("BAL-A", auctionIdBALA);
+        end.snip("LINK-A", auctionIdLINKA);
+
+        (,, usr,,,,,) = flipETHA.bids(auctionIdETHA);
+        assertTrue(usr == address(0));
+        (,, usr,,,,,) = flipBALA.bids(auctionIdBALA);
+        assertTrue(usr == address(0));
+        (,,, usr,,) = clipLINKA.sales(auctionIdLINKA);
+        assertTrue(usr == address(0));
+
+        end.skim("ETH-A", address(this));
+        end.skim("BAL-A", address(this));
+        end.skim("LINK-A", address(this));
+
+        end.free("ETH-A");
+        end.free("BAL-A");
+        end.free("LINK-A");
+
+    //     hevm.warp(block.timestamp + end.wait());
```

Comment:
@kmbarry1 I had to comment this part as even forcing a 0 surplus then we had issues on `flow`. I don't think we can properly test it without skimming CDPs and skipping/snipping auctions.

Reply:
Issues on `flow`--do you know what kind, exactly?

I am actually super worried by something I see in the magnitudes there...

Reply:
It was a safe math issue calculating the `wad` or the `fix` values.

Reply:
I see we have `rdiv(rad, rad)`, that means first arg can't be more than about 115,792 in `rad` units...but first arg is DAI of a given ilk, that's already more than 115,792 for all collaterals 😬 .